### PR TITLE
Add optional delay to perform the command on GET request

### PR DESCRIPTION
**Pull request by BarbeRouss**
My automation (Via HomeAssistant) will be able to shutdown my PC. In case of mis-configuration, or juste my wife wanting to mess with me when I'm gaming, I want to be able to abort a shutdown operation.

### CODE REVIEW

File: Karpach.RemoteShutdown.Controller/DelayForm.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            this.btnAbort.UseVisualStyleBackColor = true;
+            // 
+            // labelDelay
+            // 
+            this.labelDelay.AutoSize = true;
+            this.labelDelay.Location = new System.Drawing.Point(28, 33);
+            this.labelDelay.Name = "labelDelay";
+            this.labelDelay.Size = new System.Drawing.Size(38, 15);
+            this.labelDelay.TabIndex = 6;
+            this.labelDelay.Text = "label1";
```
Comment (karpach) [inline Karpach.RemoteShutdown.Controller/DelayForm.Designer.cs:55]: Minor, the value should be more meaningful, or empty.

File: Karpach.RemoteShutdown.Controller/DelayForm.Designer.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            // 
+            // DelayForm
+            // 
+            this.AutoScaleDimensions = new System.Drawing.SizeF(7F, 15F);
+            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
+            this.ClientSize = new System.Drawing.Size(347, 103);
+            this.Controls.Add(this.labelDelay);
+            this.Controls.Add(this.btnAbort);
+            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.None;
```
Comment (karpach) [inline Karpach.RemoteShutdown.Controller/DelayForm.Designer.cs:64]: Why System.Windows.Forms.FormBorderStyle.None? Visually it does not look good. I think it should be FixedDialog.

File: Karpach.RemoteShutdown.Controller/Helpers/HostHelper.cs [hunk 3]
```diff
@@ -75,7 +80,37 @@ internal async Task ProcessRequestAsync(string url)
...
+                {
+                    return true;
+                }
+
+                _delayForm.CommandType = commandType;
+                _delayForm.Delay = delay;
+                DialogResult dialogResult = _delayForm.ShowDialog();
+
+                if (dialogResult == DialogResult.Abort)
+                    return false;
```
Comment (karpach) [inline Karpach.RemoteShutdown.Controller/Helpers/HostHelper.cs:110]: Very minor:
{
      return false;
}

**Review by karpach — CHANGES_REQUESTED** [review]
I also use this program with Home Assistant, so this is very useful. Thank you for your contribution. Could you please address a few minor comments?

**Comment by BarbeRouss** [comment]
It should be ok now

**Review by karpach — APPROVED** [review]

**Pull request merged**